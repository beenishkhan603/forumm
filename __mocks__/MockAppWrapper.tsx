import { AuthContext } from '@libs/useAuth'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ProfileInfo } from '@type/Hooks/useAuth/Auth.type'
import ALL_MOCKS from './graphql/mocks'

export const defaultMockProfile: ProfileInfo = {
  company: '448 Studio',
  email: 'test@email.com',
  fullName: 'John Doe',
  university: '448 Studio',
  profileImageUrl: '/test.png',
  userId: '123',
  groups: [],
}

const MockAppWrapper = ({
  children,
  mocks,
  mockProfile = defaultMockProfile,
}: {
  mockProfile?: Partial<ProfileInfo>
  children: any
  mocks?: readonly MockedResponse<Record<string, any>>[] | undefined
}) => {
  const Layout = children?.type?.Layout ?? null

  const GQL_MOCKS: MockedResponse[] = ALL_MOCKS

  if (mocks) mocks.forEach((m) => GQL_MOCKS.push(m))
  return (
    <MockedProvider
      mocks={GQL_MOCKS}
      defaultOptions={{
        query: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-first',
        },
        watchQuery: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-and-network',
        },
        mutate: {
          errorPolicy: 'all',
        },
      }}
    >
      <AuthContext.Provider
        value={{
          profile: mockProfile
            ? { ...defaultMockProfile, ...mockProfile }
            : defaultMockProfile,
          setProfile: () => {},
        }}
      >
        <div id="modal-root" />
        {Layout ? <Layout>{children}</Layout> : children}
      </AuthContext.Provider>
    </MockedProvider>
  )
}

export default MockAppWrapper
