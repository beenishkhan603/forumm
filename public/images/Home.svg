<?xml version="1.0" encoding="UTF-8"?>
<svg width="512pt" height="512pt" version="1.1" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
 <path d="m415.34 210.88c12.621 12.375 19.863 29.66 19.863 47.422 0 3.5312-2.8672 6.3984-6.3984 6.3984s-6.3984-2.8672-6.3984-6.3984c0-14.34-5.8438-28.293-16.031-38.281l-120.41-118.13c-8.0703-7.9258-18.703-12.289-29.957-12.289s-21.879 4.3516-29.918 12.246c-7.4766 7.3281-95.387 93.113-120.19 117.11-10.355 10.016-16.293 24.066-16.293 38.539 0 3.5312-2.8672 6.3984-6.3984 6.3984-3.5352 0.003906-6.4023-2.8633-6.4023-6.3984 0-17.922 7.3633-35.324 20.191-47.742 24.797-23.977 112.66-109.73 120.12-117.04 10.441-10.254 24.254-15.906 38.883-15.906s28.453 5.6641 38.926 15.953zm-32.461 33.625c3.6758 3.6055 5.7812 8.6328 5.7812 13.793v145.84c0 17.133-14.551 31.062-32.441 31.062h-61.066c-3.5312 0-6.3984-2.8672-6.3984-6.3984l-0.003906-89.805c0-16.285-13.891-29.699-30.965-29.91-8.9609-0.097657-17.105 3.1445-23.199 9.1641-5.5508 5.4766-8.6055 12.66-8.6055 20.215v90.336c0 3.5312-2.8672 6.3984-6.3984 6.3984h-63.793c-17.891 0-32.445-13.938-32.445-31.062v-146.65c0-5.207 2.1406-10.266 5.8711-13.879 25.309-24.48 115.13-112.14 120.35-117.25 3.543-3.4805 9.3203-3.4922 12.875 0.003906zm-7.0195 13.797c0-1.7461-0.71094-3.4414-1.9453-4.6523l-117.91-115.68c-15.789 15.426-94.457 92.188-117.88 114.84-1.2539 1.2148-1.9766 2.9219-1.9766 4.6836v146.65c0 10.07 8.8125 18.262 19.645 18.262h57.387v-83.941c0-11.004 4.4102-21.418 12.41-29.32 8.5547-8.4414 20.102-12.914 32.352-12.855 24.043 0.29297 43.605 19.449 43.605 42.711l0.003906 83.402h54.668c10.828 0 19.641-8.1914 19.641-18.262z" />
</svg>
