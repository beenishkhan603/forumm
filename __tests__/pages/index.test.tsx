import { render } from '@testing-library/react'
import Home from 'pages'
import MockAppWrapper from '__mocks__/MockAppWrapper'

xit('renders index page unchanged', () => {
  //   const { container } = render(
  //     <MockAppWrapper>
  //       <Home />
  //     </MockAppWrapper>
  //   )
  //   expect(container).toMatchSnapshot()
})
