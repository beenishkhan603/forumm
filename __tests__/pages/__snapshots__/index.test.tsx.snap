// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders index page unchanged 1`] = `
<div>
  <div
    id="modal-root"
  />
  <div
    class="overflow-hidden min-h-screen flex flex-col "
    style="background-color: rgb(255, 255, 255); color: rgb(24, 26, 32);"
  >
    <header>
      <div
        class="flex px-4 md:px-20 lg:px-[156px] py-3 justify-between items-center h-[81px]"
        style="color: rgb(24, 26, 32); background-color: rgb(243, 246, 252);"
      >
        <div
          class="flex space-x-4 items-center cursor-pointer"
          style="color: rgb(24, 26, 32);"
        >
          <div
            class="flex"
            style="color: rgb(24, 26, 32);"
          />
          <span
            class="select-none"
          />
          <div
            class="relative flex-shrink-0  cursor-pointer"
          >
            <img
              alt="Profile Image"
              class="w-10 h-10 bg-contain bg-center rounded-full object-cover "
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="http://localhost/_next/image?url=%2Ftest.png&w=64&q=75"
              srcset="/_next/image?url=%2Ftest.png&w=32&q=75 1x, /_next/image?url=%2Ftest.png&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
          </div>
        </div>
      </div>
      <div
        class="z-50 w-64 absolute px-6 right-0 mr-5 rounded-b flex flex-col space-y-2 text-lg transition-all h-0 overflow-hidden"
        style="background-color: rgb(243, 246, 252); color: rgb(24, 26, 32);"
      >
        <div
          class="cursor-pointer flex items-center"
          style="color: rgb(24, 26, 32);"
        >
          <svg
            class="inline w-5 h-6 mr-2"
            fill="#181A20"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
            />
          </svg>
          User Profile
        </div>
        <div
          class="cursor-pointer flex items-center"
          style="color: rgb(24, 26, 32);"
        >
          <svg
            class="inline w-5 h-6 mr-2 text-181A20"
          />
          Support
        </div>
        <div
          class="cursor-pointer flex items-center"
          style="color: rgb(24, 26, 32);"
        >
          <svg
            class="inline w-5 h-6 mr-2"
            fill="#181A20"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M17 8l-1.41 1.41L17.17 11H9v2h8.17l-1.58 1.58L17 16l4-4-4-4zM5 5h7V3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h7v-2H5V5z"
            />
          </svg>
          Logout
        </div>
      </div>
    </header>
    <div
      class="flex-1 transition-all"
      style="background-color: rgb(255, 255, 255);"
    >
      <div
        class="flex justify-center"
        style="background-color: rgb(255, 255, 255); color: rgb(24, 26, 32);"
      >
        <div
          class="p-2 py-6 sm:p-6 max-w-6xl flex-grow"
          style="color: rgb(24, 26, 32);"
        >
          <svg
            class="w-44 h-16 -ml-1"
          />
          <div
            class="text-2xl mb-2"
            style="color: rgb(24, 26, 32);"
          >
            Welcome to Forumm.
          </div>
          <div
            class="my-8 sm:w-3/4 lg:w-1/2 text-sm"
            style="color: rgb(24, 26, 32);"
          >
            Create immersive virtual, hybrid, and in-person event experiences for your audience, no matter where they are.
          </div>
          <div
            class="text-2xl mb-2"
            style="color: rgb(24, 26, 32);"
          >
            Open Events
          </div>
          <div
            class="my-2 sm:w-3/4 lg:w-1/2 text-sm"
            style="color: rgb(24, 26, 32);"
          >
            Check out our upcoming, and open to all events here.
          </div>
          <div
            style="color: rgb(24, 26, 32);"
          >
            <div
              class="w-full h-full justify-center items-center flex"
              data-test-id="box-spinner"
              style="color: rgb(24, 26, 32);"
            >
              <svg
                class="w-12 h-12 animate-spin block "
                data-testid="loading"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex w-full min-h-[200px] bg-forumm-blue-dark p-16 justify-center"
      style="color: rgb(24, 26, 32);"
    >
      <div
        class="flex flex-col sm:w-1/3 justify-center text-center"
        style="color: rgb(24, 26, 32);"
      >
        <div
          class="flex w-full justify-center"
          style="color: rgb(24, 26, 32);"
        >
          <svg />
          <svg
            class="mr-2 mt-1.5"
          />
        </div>
        <div
          class="mt-4"
          style="color: rgb(24, 26, 32);"
        >
          <div
            class="text-sm !text-white font-normal"
            style="color: rgb(24, 26, 32);"
          >
            Use the power of global events to fuel your alumni donations
          </div>
        </div>
        <div
          class="mt-6 flex flex-row justify-center text-center"
          style="color: rgb(24, 26, 32);"
        >
          <a
            href="https://www.linkedin.com/company/68805696/admin/feed/posts/"
            target="_blank"
          >
            <svg
              class="mr-3 cursor-pointer"
            />
          </a>
          <a
            href="https://twitter.com/ForummOfficial"
            target="_blank"
          >
            <svg
              class="mr-3"
            />
          </a>
          <a
            href="https://www.facebook.com/forummofficial/"
            target="_blank"
          >
            <svg
              class="mr-3"
            />
          </a>
          <a
            href="https://www.instagram.com/forummofficial/"
            target="_blank"
          >
            <svg />
          </a>
        </div>
        <div
          class="mt-4"
          style="color: rgb(24, 26, 32);"
        >
          <div
            class="text-md !text-white font-regular"
            style="color: rgb(24, 26, 32);"
          >
            Â© Forumm 
            2024
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
